pkg/edition/java/proto/packet/login.go

```diff
@@ -84,7 +85,7 @@ func (l *LoginPluginResponse) Encode(_ *proto.PacketContext, wr io.Writer) (err
 	if err != nil {
 		return err
 	}
-	return util.WriteBytes(wr, l.Data)
+	return util.WriteRawBytes(wr, l.Data)
```

Comment:
velocity requires a non-length-prefixed byte[] at the end of the packet. idk, why, but it barfs when it receives a length prefix.